[feature][bugfix] support reloading fluentd

trombik opened the pull request:
fixes #44 
fixes #40 

* introduce fluentd_log_file
* remove `23` postfix in paths of `fluentd` binaries in OpenBSD

Commit:
add spec to check reloading

also, enable `debug` log to log received HUP signal

Commit:
introduce fluentd_log_file

Commit:
enable reload in OpenBSD

Commit:
patch rc.d script to enable reload in FreeBSD

Commit:
patch rc.d script to enable reload in FreeBSD

Commit:
add Reload fluentd handler

Commit:
pass fluentd_flags to arguments only when OpenBSD

Commit:
use fluentd_log_file, instead of fluentd_log_dir

also use the same flags in without_ca_key to enable debug

Commit:
insert "pause" between restart and reload handlers

fixes #44

Commit:
typo

Commit:
s/23//g, symlinks has been created by the role

Commit:
update README

Commit:
remove tests of service reload

in each play, the service is reloaded and there should always be reload
event in the log. should reload command fail, the play also fails. as
such, the tests are useless.

mitsururike commented on tasks/install-FreeBSD.yml:
```diff
@@ -4,6 +4,14 @@
+  register: register_fluentd_rubygem_fluentd
+
+- name: Patch rc.d script
+  patch:
```
lineinfile is safer as the rc.d script might be modified in the other version of the package and the patch fails.

trombik replied:
but by the time the system is deployed, we should be using our own package tree. it's a hack from the start and i would like to keep the patch for inclusion.

mitsururike replied:
it should be written as a comment.

trombik replied:
fixed in b03e8b2

Commit:
add XXX

https://github.com/reallyenglish/ansible-role-fluentd/pull/45#discussion_r118859816
